### PR TITLE
Remove import of defaultdict from compat/utils

### PR DESCRIPTION
**Pull request by agamrafaeli**

### CODE REVIEW

**Comment by zzzeek** [comment]
nice!

**Comment by zzzeek** [comment]
note I am doing an epic module import refactor at the moment as well to do correct import ordering and enforcing flake8 in CI, this can likely go in before that.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work to try to get revision 23bee91f009c9e064e39025b26e7717004a90391 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 23bee91f009c9e064e39025b26e7717004a90391: https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1064

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

Code-Review+2

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

-Code-Review

looks like this missed at least one import of util.defaultdict in sqlalchemy.testing

**Comment by agamrafaeli** [comment]
@zzzeek pushed a fix. Can I help with this massive refactor? Any way to offload some of the tasks?

**Comment by zzzeek** [comment]
> @zzzeek pushed a fix. Can I help with this massive refactor? Any way to offload some of the tasks?

I've been working through using a tool I wrote called [zimports](http://github.com/sqlalchemyorg/zimports/) which mostly works, as well as a pretty slick code formatting tool [black](https://github.com/ambv/black) to reformat everything, and to that end a full pep8/flake8 job is turned on in jenkins to run it.    It's close to being done at https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1062/, however if you want a real job, the same thing needs to be done for in theory all the other sqlalchemy projects, the most important being Alembic.    the flake8 settings I'm working with are [here](https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1062/23/setup.cfg) and the plugins I'm using are [here](https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1062/23/tox.ini).    so you say "tox -e pep8" and watch it go.   if you were interested in working on that, I'd welcome that.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work to try to get revision d42d616363b5054e483e679cc38e14b878066eb7 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset d42d616363b5054e483e679cc38e14b878066eb7 added to existing Gerrit review https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1064

**Review by sqla-tester — COMMENTED** [review]
**mike bayer** (zzzeek) wrote:

(1 comment)

before you push, if possible, just run the tests quick:

    python setup.py test

with a py2k interpreter.  if you don't have py2k lying around then no biggie.

File: lib/sqlalchemy/orm/unitofwork.py [hunk 1]
```diff
@@ -13,11 +13,12 @@
 
 """
 
+import collections
```
Comment (sqla-tester) [inline lib/sqlalchemy/orm/unitofwork.py]: **mike bayer** (zzzeek) wrote:

so there is a module called sqlalchemy.orm.collections.   so to make this work in Python 2.7, you need to add "from __future__ import absolute_import" as the first import in the file.

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

this PR seems to be dead, feel free to re-submit if you have time thanks!

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/1064 has been **abandoned**.  That means that at least for the moment I need to close this pull request. Sorry it didn't work out :(

**Pull request closed without merging**